### PR TITLE
JSON schemas - schemas

### PR DESCRIPTION
**Pull request by Adrijaned**
This PR introduces some JSON schemas for validating integrity and sanity of module JSONs. This does not integrate the schemas with DestinationSol in any way, that is to be done in a different PR.

### CODE REVIEW

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by GooeyHub** [comment]
Uh oh, something went wrong with the build. Need to check on that

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

File: engine/src/main/resources/assets/schemas/schema-hull.json [hunk 1]
```diff
@@ -0,0 +1,308 @@
...
+      "type": "string",
+      "description": "Id of the engine on the ship",
+      "pattern": "^\\w+:\\w+$"
+    },
+    "ability": {
+      "type": "object",
+      "properties": {
+        "type": {
+          "type": "string",
+          "description": "Type of ship's ability - one of 'sloMo', 'knockBack', 'teleport', 'emWave', 'unShield'"
```
[the remark this answers is not in the document]
Comment (Adrijaned) [reply engine/src/main/resources/assets/schemas/schema-hull.json]: The enum is then specified in the individual "anyOf" cases

File: engine/src/main/java/org/destinationsol/game/ship/ShipBuilder.java [hunk 2]
```diff
@@ -70,30 +69,7 @@ public ShipBuilder() {
...
-                poly.getVertex(i, v);
-                if (v.x < lowestX) {
-                    base = f;
-                    lowestX = v.x;
-                }
-            }
-        }
-        SolMath.free(v);
-        return base;
+    private static void getBase() {
```
[the remark this answers is not in the document]
Comment (Adrijaned) [reply engine/src/main/java/org/destinationsol/game/ship/ShipBuilder.java]: Missed it.

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by SkySom** [comment]
Assigning over to @vampcat for final handling but all is good on myside

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Comment by GooeyHub** [comment]
Uh oh, something went wrong with the build. Need to check on that

**Comment by Adrijaned** [comment]
Closed in favor of #371 (everything else but contents of that PR has already been included in other PRs)

**Pull request closed without merging**